### DISCUSSION TITLE
Gradle wrapper is missing the executable bit

### DISCUSSION THREAD

**Discussion by stefan-g** [renovatebot/renovate#28763] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I got the warning "Gradle wrapper is missing the executable bit" when the renovate-runner is trying to update gradle of a project.

i'm using the latest version of self-hosted renovate.

Any hint what i have to change? Where does the runner tried to execute the gradle wrapper?

### Logs (if relevant)

<details><summary>Logs</summary>

```

 WARN: Gradle wrapper is missing the executable bit (repository=renovate/renovate-spring-boot-test, branch=renovate/gradle-8.x)
DEBUG: Updating gradle wrapper: "./gradlew :wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-8.7-bin.zip" (repository=renovate/renovate-spring-boot-test, branch=renovate/gradle-8.x)

```

</details>

**Comment by Churro** [comment]
Does the update work correctly after this warning? You can get rid of this by setting the executable bit, e.g., `chmod +x ./gradlew`.

Nonetheless, it might make sense to lower the log level from `WARN` to `DEBUG` since renovate has a workaround for such cases:
https://github.com/renovatebot/renovate/blob/db1e79cf164ae3466113bb5456d4cc0a15951d12/lib/modules/manager/gradle-wrapper/utils.ts#L30-L34

**Reply by stefan-g** [reply]
The update does only create a merge request for  the changes in gradle/wrapper/gradle-wrapper.properties. gradlew, gradlew.bat and gradle-wrapper.jar are not updated. 

where do i have to set the executable bit? i cloned the renovate runner project from here https://gitlab.com/renovate-bot/renovate-runner and let it run like this.

**Reply by Churro** [reply]
renovate calls `gradlew` in the project you are running it on, so the bit needs to be set there. As said, there is a workaround in renovate that attempts setting the bit on-the-fly. The warning doesn't abort the update, so any errors that could lead to PRs with missing wrapper binaries are denoted at different places in the log.

**Reply by stefan-g** [reply]
when i run the renovate-runner a second time. the warnings are gone but still only the version in  gradle/wrapper/gradle-wrapper.properties is updated. is there an additional setting in the renovate.json required? it's nothing mentioned here https://docs.renovatebot.com/java/#gradle-wrapper

I have seen in the logs that there was an other error in the logs (does after i run the runner the second time):
```
"cmd": "/bin/sh -c ./gradlew :wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-8.7-bin.zip",
         "stderr": "\nFAILURE: Build failed with an exception.\n\n* Where:\nSettings file '/builds/renovate-runner/renovate/repos/gitlab/renovate/renovate-spring-boot-test/settings.gradle' line: 7\n\n* What went wrong:\nA problem occurred evaluating settings 'renovate-spring-boot-test'.\n> Could not get unknown property 'NEXUS_URL'
```
Env variable should be there (at last build pipeline runs without error in the test project)

**Reply by Churro** [reply]
renovate doesn't consume env vars from the build jobs but since you're running self-hotsed, you can supplement `NEXUS_URL` in your config.js, not renovate.json -> https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables

**Reply by stefan-g** [reply]
It worked with the settings in the config.js but it was a little bit strange because it took several attempts of the renovate-runner until "command": "./gradlew :wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-8.7-bin.zip" was executed again. before only the version in the gradle-wrapper.properties was updated.

**Reply by rarkins** [reply, marked as answer]
https://github.com/renovatebot/renovate/pull/28844
